### ISSUE TITLE
Amplify cognito auth don't work propetly when is IPv6 automatically

### ISSUE THREAD

**Issue by IsmetGlumcevic** [aws-amplify/amplify-js#10875]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Next.js

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
nextjs 12

amplify -v 9.1.0

```

</details>

### Describe the bug

Hi, i have installed touch VPN on the browser and Cognito auth doesn't work properly. When i enable VPN its works, sometimes fast sometimes slow. When are disabled doesn't work. When using mobile internet on a laptop works well. 
My code in react component is good. Is there a way to fix this isssue. I remove touch vpn exenstion but signin don't work. Only work with an enabled VPN.

### Expected behavior

The expected behavior is that auth work properly because we don't know if the user has installed an extension for VPN.

### Reproduction steps

Install the touch VPN extension in chrome
Install and setup amplify auth in react or next js project
Try to signup or signin.

package.json

```

{
  "name": "xxx",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@aws-amplify/ui-react": "^3.3.0",
    "@elastic/elasticsearch": "7.13.0",
    "@tanstack/react-query": "^4.16.1",
    "@tanstack/react-query-devtools": "^4.16.1",
    "@tanstack/react-table": "^8.7.4",
    "aws-amplify": "^4.3.30",
    "axios": "^1.2.1",
    "jsonp": "^0.2.1",
    "next": "12.2.4",
    "query-string": "^8.1.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.41.0",
    "react-select": "^5.4.0",
    "react-toastify": "^9.1.1",
    "request-ip": "^3.3.0",
    "swiper": "^8.4.4",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.8",
    "eslint": "8.21.0",
    "eslint-config-next": "12.2.4",
    "postcss": "^8.4.16",
    "tailwindcss": "^3.1.8"
  }
}

```

aws-exports.js

```

const awsmobile = {
    "aws_project_region": "eu-central-1",
    "aws_cloud_logic_custom": [
        {
            "name": "xxx",
            "endpoint": "https://xxx.execute-api.eu-central-1.amazonaws.com/localy",
            "region": "eu-central-1"
        }
    ],
    "aws_dynamodb_all_tables_region": "eu-central-1",
    "aws_dynamodb_table_schemas": [
        {
            "tableName": "xxx-xxx-localy",
            "region": "eu-central-1"
        }
    ],
    "aws_user_files_s3_bucket": "xxx-localy",
    "aws_user_files_s3_bucket_region": "eu-central-1",
    "aws_cognito_identity_pool_id": "eu-central-1:xxx",
    "aws_cognito_region": "eu-central-1",
    "aws_user_pools_id": "eu-central-1_xxx",
    "aws_user_pools_web_client_id": "xxx",
    "oauth": {
        "domain": "xxx.auth.eu-central-1.amazoncognito.com",
        "scope": [
            "aws.cognito.signin.user.admin",
            "email",
            "openid",
            "phone",
            "profile"
        ],
        "redirectSignIn": "http://localhost:3000",
        "redirectSignOut": "http://localhost:3000",
        "responseType": "code"
    },
    "federationTarget": "COGNITO_USER_AND_IDENTITY_POOLS",
    "aws_cognito_username_attributes": [
        "EMAIL"
    ],
    "aws_cognito_social_providers": [
        "FACEBOOK",
        "GOOGLE"
    ],
    "aws_cognito_signup_attributes": [
        "EMAIL"
    ],
    "aws_cognito_mfa_configuration": "OFF",
    "aws_cognito_mfa_types": [],
    "aws_cognito_password_protection_settings": {
        "passwordPolicyMinLength": 8,
        "passwordPolicyCharacters": [
            "REQUIRES_LOWERCASE",
            "REQUIRES_UPPERCASE",
            "REQUIRES_NUMBERS",
            "REQUIRES_SYMBOLS"
        ]
    },
    "aws_cognito_verification_mechanisms": [
        "EMAIL"
    ]
};


```

**Comment by IsmetGlumcevic** [comment]
Hi @tannerabread
I update the details.
I try firefox and get the same issue. On mobile internet  4g/5g works, VPN sometimes works, but on local internet without VPN doesn't work. I also use react native and get the same issue.

**Comment by IsmetGlumcevic** [comment]
<img width="942" alt="Screenshot 2023-01-19 at 06 51 38" src="https://user-images.githubusercontent.com/14037875/213367041-04127752-481a-49a0-9a17-76f96a0fa7ea.png">

**Comment by IsmetGlumcevic** [comment]
<img width="1048" alt="Screenshot 2023-01-19 at 07 00 28" src="https://user-images.githubusercontent.com/14037875/213367229-c26b406b-a0f8-41dd-b7b6-e74c682467c4.png">

**Comment by IsmetGlumcevic** [comment]
@tannerabread I find a reason for the issue, when I change IPv6 to link-local only it works well, but when is IPv6 automatically doesn't work. How IPv6 affects Cognito auth?

**Comment by tannerabread** [comment]
@IsmetGlumcevic 
I couldn't find any documentation from the Cognito team about this issue and have asked internally for more clarification. I will get back to this issue when I have more information

**Comment by tannerabread** [comment]
@IsmetGlumcevic 
After speaking more with the team, there are a few things we should try for troubleshooting before looking to Cognito as the cause.

Can you check if you can access either of the following endpoints while under your `automatically` configured IPv6 settings:
https://ipv6.google.com/ 
https://[2a05:d014:32e:700:187b:4df3:d16f:9e17]

The 2nd link there is the IPv6 of the domain in your screenshot

**Comment by IsmetGlumcevic** [comment]
@tannerabread 
I can accses https://ipv6.google.com/
I cannot access the second link, get error: Your connection is not private.

**Comment by tannerabread** [comment]
Hi @IsmetGlumcevic

I tried to reproduce this and ran into some errors when my `Configure IPv4` settings were set to "Off", but I received a different error than you. When I turned my `Configure IPv4` to "Using DHCP", all of the requests succeeded with every IPv6 

Do you have anything selected under `Configure IPv4`? The "Using DHCP" option was the default for me

**Comment by IsmetGlumcevic** [comment]
Hi @tannerabread 

Configured IPv4 is "Using DHCP". When I use wifi from the phone hotspot it works well with every IPv6.  

Is there an issue somewhere else? When I log in from a mobile browser the result is the same as on a desktop.

**Comment by tannerabread** [comment]
Hi @IsmetGlumcevic, maybe I originally misunderstood you. I thought it was only failing on your Mac and was working everywhere else. If it is failing on multiple networks, can you share more about your project and how you are using the amplify library so we can debug the project itself?

The "Network Error" being returned would point to something with your network setup, but I wouldn't be able to troubleshoot your network. I could suggest trying it on another network (if you are able to) and seeing if you get the same results.

**Comment by tannerabread** [comment]
@IsmetGlumcevic I ran into the "network error" problem once when trying to use an amplify project where the backend had been deleted. Just verifying that this still works sometimes but not others? For me, it wouldn't work at all

**Comment by jacobira** [comment]
I am experiencing a similar issue with my React Native app using Amplify Cognito authentication. Whenever attempting authentication using the `Auth.signIn()` method while connected to my home wifi network (has IPv6 enabled) the application sits endlessly waiting on a response from Cognito. When I switch to mobile data, mobile data hotspot, or use a VPN, it works successfully. I should also note that this has **only been happening on Android** devices, both emulators and physical devices.
It has been exceptionally hard to gather data on this, since connecting to a debugger, essentially piggy-backing on a browser, allows everything to work just fine and thus cannot reproduce the issue while debugging.
Current specs:
"react-native": "^0.70.0"
"aws-amplify": "^5.0.12"
"amazon-cognito-identity-js": "^5.2.10"

**Comment by jacobira** [comment]
@tannerabread 
It was working until recently, though the last time I did any android testing (as this app is still in the pre-launch/development phase) was about a month ago.

**Comment by jacobira** [comment]
I went ahead and used Network Inspect on Android Studio, and as you can see no network requests even make it out the gate. The highlighted portion is when I kick off the `Auth.signIn()` function. The first with nothing is when connected to WiFi, the second is when using mobile 5G.
![Screen Shot 2023-02-01 at 2 47 19 PM](https://user-images.githubusercontent.com/35573298/216171431-f39836f3-ac42-476b-9261-47e686710fda.png)
![Screen Shot 2023-02-01 at 2 49 25 PM](https://user-images.githubusercontent.com/35573298/216171437-2c0c7255-298c-4477-83a7-0ad25c853420.png)

**Comment by jacobira** [comment]
@tannerabread 
After switching to 5G, I'm able to authenticate. Post-authentication, I turn WiFi back on and am immediately seeing graphql hits going through correctly, indicating that there shouldn't be anything wrong with my network.
![Screen Shot 2023-02-01 at 2 58 45 PM](https://user-images.githubusercontent.com/35573298/216173151-52be03d9-706f-413b-8826-3eec67224fa3.png)

**Comment by tannerabread** [comment]
I'm not seeing those same results, I tried on another computer just to test it out and have no problems with Auth.signIn

I will have someone else test it

**Comment by jacobira** [comment]
@tannerabread 
This morning I went through a full factory reset of my router and to my great relief with IPv6 enabled I am getting through, at least for now. However, the fact that this behavior began at all is quite interesting. I'm not a netcomms pro, but I have to wonder if modern IPv6 equipment has certain safeguards that are getting triggered by the number of hits a developer makes to aws-amplify authentication endpoints.
So in short... consider my issue resolved, though I'm still generally suspicious of what started it. Not sure if you guys have the time to bombard an IPv6 router with endpoint calls to see if it triggers a firewall or anything. If you do, please let me know as I would be curious to know what happens.
I can't speak for @IsmetGlumcevic's issue, but I would suggest he do a full router reset and see if that yields any results.

**Comment by tannerabread** [comment]
@jacobira I understand your concern. I think this would be more related to Cognito itself but I will see if @IsmetGlumcevic has anything more on this issue. The reason I think it would be more Cognito related and not Amplify library related is because we use Cognito as the Auth provider and your router reset solved the issue without any changes to the app.

I will also monitor cognito related internal channels for any mention of this issue. I'm curious as well but still haven't been able to reproduce this across 3 different computers in my home running IPv6 on my network.

I won't close this yet, but if I do and you start experiencing this again just comment back on here and I will reopen it.

**Comment by IsmetGlumcevic** [comment]
@tannerabread I reset my router and auth work for 1 hour. After 1 hour gets the same error. In debug console see there is token expiration of 1 hour. I tested on different routers and networks and get the same error on web browser and react native app. Issues exist in production not only in development.

This is a fresh project with nextjs and amplify on vercel.

https://profile-app-self.vercel.app/
https://github.com/IsmetGlumcevic/profilea-pp

**Comment by jacobira** [comment]
@tannerabread just like @IsmetGlumcevic authentication only worked for a short period after resetting my router yesterday.

**Comment by jacobira** [comment]
@tannerabread I've disabled IPv6 on my personal router and everything is going through. I'll leave it disabled for the time being, but this really seems like it could be a major issue since we have no control at all what internet source a client is using and whether it utilizes IPv6.
Is there any way for you to get the Cognito team involved without @IsmetGlumcevic and I having to rehash everything on a new thread under their project?

**Comment by tannerabread** [comment]
@jacobira @IsmetGlumcevic I will investigate further and get cognito involved

**Comment by tannerabread** [comment]
@jacobira @IsmetGlumcevic Since this seems to be an isolated issue, as I'm sure there are many users using IPv6 with no trouble, can someone share how they are using the library within their app so we can rule out misconfigured apps?

**Comment by IsmetGlumcevic** [comment]
@tannerabread  This simple example gives me this issue. I tested on macOS, ios, and android. My home router is zte model zxhn h108n.

https://profile-app-self.vercel.app/
https://github.com/IsmetGlumcevic/profilea-pp
username: user1
password: Amplify123

**Comment by tannerabread** [comment]
Sorry @IsmetGlumcevic I missed you linking that repo previously.

I don't currently see anything wrong with the config of your app, and I assume with this simple example the resources you set up were fairly basic? The only thing I have found so far is [IPv6 considerations](https://docs.aws.amazon.com/vpn/latest/clientvpn-admin/ipv6-considerations.html) if you have a Client VPN service set up.

Can you confirm if you created your AWS resources with minimal/default settings?

I tried your app on my network on 2 computers and am still unable to replicate the issue.

**Comment by IsmetGlumcevic** [comment]
@tannerabread I remove Client VPN service, and I created your AWS resources with minimal/default settings.

**Comment by tannerabread** [comment]
@IsmetGlumcevic @jacobira 
Are either of you still experiencing this? I know problems don't always go away on their own but I still haven't been able to reproduce this and haven't currently gotten much information back from Cognito. From what I've read, IPv6 should work with Cognito calls. I was hoping that maybe it was some spotty service that might have resolved itself.

It's curious that your resetting your router settings fixed it for such a short period of time, and I'm trying to figure out how to proceed with this issue. I will try to talk to the Cognito team again but is there anything else either of you can think of that we are missing for reproducing this issue?

**Comment by IsmetGlumcevic** [comment]
Hi @tannerabread, after changing the router it works great.
